### ISSUE TITLE
Unexpected statement::bind_param behavior when nulls parameter is nullptr

### ISSUE THREAD

**Issue by mloskot** [nanodbc/nanodbc#104]
_From @AndrewJD79 on April 7, 2016 1:37_

Hi

statement class has several overloads of bind method:
1. `void bind(short, const T*, std::size_t , param_direction);`
2. `void bind(short, const T*, std::size_t , const bool* nulls, param_direction direction);`

I've faced with unexpected behavior of  second function. If nulls parameter is equal to nullptr then implementation binds parameter as NULL value. This happens because of following code:
statement::statement_impl::bind()
        `if((null_sentry && !equals(values[i], *null_sentry)) || (nulls && !nulls[i]))`
           `bind_len_or_null_[param][i] = parameter_size;`

Maybe it's by design but I would expect that providing nullptr means 'I don't care about null' and this overload is just more generalized version of first function.
i.e. bind(0, &x, 1, nullptr, PARAM_IN) is equal to bind(0, &x, 1, PARAM_IN);

_Copied from original issue: lexicalunit/nanodbc#122_

**Comment by mloskot** [comment]
_From @lexicalunit on May 5, 2016 11:7_

@AndrewJD79 I agree with you. The intent was for one overload to be a more generalized version of the other. I've created PR #140 which should resolve the issue and adds unit tests to make sure we don't get a regression on this feature. Thanks for the detailed bug report! 👍

I'll close this topic only once the PR's CI builds are all green and it's merged into master.
